testsuite/features/support/remote_node.rb

```diff
@@ -46,7 +46,8 @@ def initialize(host, port: 22)
     raise StandardError, "No FQDN for '#{@hostname}'. Response code: #{code}" if @full_hostname.empty?
 
     $stdout.puts "Host '#{@host}' is alive with determined hostname #{@hostname} and FQDN #{@full_hostname}" unless $build_validation
-    @os_version, @os_family = get_os_version
+    @os_version, @os_family = get_os_version(real_host: false)
+    @real_host_os_varsion, @real_host_os_family = get_os_version(real_host: true)
```

Comment:
```suggestion
if @has_mgrctl
    @container_os_version, @container_os_family = get_os_version(run_in_container: true)
    @os_version, @os_family = get_os_version(run_in_container: false)
else
    @os_version, @os_family = get_os_version(run_in_container: false)
end    
```

Reply:
1- This test is already done in get_os_version
2 - some variables are not set in one of the cases

But I agree it's more legible

Reply:
Functionally equivalent, with more risk, rejected, even if Paolo kind of liked the suggestion

Reply:
I don't see more risk, and I see it better from an OOP design. Still, as said, your call to implement it as you like, because I can't commit to track these changes or help on stabilizing them.